step-release-vis/src/app/components/environments/environments.ts

```diff
@@ -26,39 +27,46 @@ export class EnvironmentsComponent implements OnInit {
   startTimestamp: number; // current start timestamp
   endTimestamp: number; // current end timestamp
 
-  envJson: string;
+  dataFound: boolean;
   timelinePointsAmount: number;
   timelinePoints: TimelinePoint[];
 
   constructor(
-    private fileService: FileService,
+    private dataService: DataService,
     private candidateService: CandidateService,
     private protoBufferService: ProtoBufferService
   ) {}
 
   ngOnInit(): void {
-    // this.readJson();
-    this.readBinary();
+    this.readProtoBinaryData();
```

Comment:
Should I add a possibility to specify the type of data used as GET params? Smth like `?dataSource=binary`

Reply:
GET has size limits lower than what our data would be if we base64 encode (and other case the format would not be OK) so i think we can skip that

Reply:
No, I mean just an enum type of input the user wants, so I'll have
```
if (query.dataSource == 'binary') {
  readBinary
} else if ...
```

Reply:
Because at this point you can submit both json and binary, but it'll still only look for binary. You have to change the code to read from json.